tests/helper.py

```diff
@@ -295,6 +297,22 @@ def flair_diff(root, other):
     return list(root_items - other_items)
 
 
+def mock_sys_stream(streamname, defaulttext=None):
+    def wrapper(f):
+        @wraps(f)
+        def wrapped(obj):
+            stream = cStringIO()
+            setattr(sys, streamname, stream)
+            if defaulttext is not None:
+                stream.write(defaulttext)
+                stream.seek(0)
+            retval = f(obj)
+            setattr(sys, streamname, getattr(sys, "__{}__".format(streamname)))
```

Comment:
This should probably be in a finally block in the event the function call fails. The original code had the same issue.